Sources/StreamChat/Repositories/MessageRepository.swift

```diff
@@ -90,22 +90,22 @@ class MessageRepository {
     func saveSuccessfullySentMessage(
         cid: ChannelId,
         message: MessagePayload,
-        completion: @escaping (ChatMessage?) -> Void
+        completion: @escaping (Result<ChatMessage, Error>) -> Void
     ) {
-        var messageModel: ChatMessage?
         database.write({
             let messageDTO = try $0.saveMessage(payload: message, for: cid, syncOwnReactions: false, cache: nil)
             if messageDTO.localMessageState == .sending || messageDTO.localMessageState == .sendingFailed {
                 messageDTO.locallyCreatedAt = nil
                 messageDTO.localMessageState = nil
             }
 
-            messageModel = try? messageDTO.asModel()
+            let messageModel = try messageDTO.asModel()
+            completion(.success(messageModel))
```

Comment:
why is it moved here?

Reply:
Here we garantee the message is always present 👍

Reply:
Because we now have a success and error branches, there's no need to keep a reference of this model, and return an optional value.